BankClient/Bank.cs

```diff
@@ -1,166 +1,115 @@
+        
+        if (filterList.Count != 0)
         {
-            Console.WriteLine("Nobody was found");
-            return;
+            var oneUserInformation = filterList[0];
+            oneUserInformation.Print();
+            return oneUserInformation;
         }
-        Console.WriteLine($"Firstname: {filterlist[0].FirstName}, LastName: {filterlist[0].LastName}, Phonenumber: {filterlist[0].Phonenumber}, userID: {filterlist[0].ID}");
-    }
+        return null;
 
-    public void ShowAllUsers()
+    }
+    /// <summary>
+    /// Search user by ID
+    /// </summary>
+    /// <returns>Return user or null</returns>
+    /// 
+    public UserInformation GetUserById(Guid id)
     {
-        if (Users != null)
+        var result = Users.FirstOrDefault(x => x.Id.ToString() == id.ToString());
+        if (result != null)
         {
-            int i = 1;
-            foreach (var user in Users)
-            {
-                Console.WriteLine(
-                    $"User {i}: Firstname: {user.FirstName}, LastName: {user.LastName}, Phonenumber: {user.Phonenumber}, userID: {user.ID}");
-                    i++;
-            }
-        }
-        else
-        {
-            Console.WriteLine("There are not users in the list");
+            result.Print();
         }
+    
+        return result;
     }
-    public void ChangeUserInfomation()
+
+    /// <summary>
+    /// Change user information
+    /// </summary>
+    /// <param name="id">Takes <see cref="UserInformation"/> object which has all user information</param>
+    /// <returns>Nothing</returns>
+    public void ChangeUserInformation(Guid id)
     {
-        Console.Write("Enter userID: ");
-        var id = Console.ReadLine().Trim();
-        bool flag = false;
-        foreach (var user in Users)
-        {
-            if (user.ID.ToString() == id)
-            {
-                flag = true;
-                Console.Write("Firstname: ");
-                user.FirstName = Console.ReadLine();
-                Console.Write("Lastname: ");
-                user.LastName = Console.ReadLine();
-                Console.Write("Phonenumber: ");
-                user.Phonenumber = Console.ReadLine();
-                Console.WriteLine("The client was changed");
-                break;
+        for (var i = 0; i < Users.Count; i++)
+        { 
+            if (Users[i].Id.ToString() == id.ToString()) 
+            {  
+                var changedUser = UserInterface.GetInformationFromUser();
+                Users[i] = changedUser;
             }
         }
-
-        if (flag == false)
-            Console.WriteLine("User wasn't found");
     }
 
+    /// <summary>
+    /// Add new user in the list
+    /// </summary>
+    /// <returns>Nothing</returns>
     public void AddUserInformation()
     {
-        Users.Add(new UserInformation());
+        var newUser = UserInterface.GetInformationFromUser();
+        Users.Add(newUser);
     }
```

Comment:
This method should accept UserInformation as an argument and return created UserInformation

Reply:
I agree, this method should return `UserInformation`.
I don't really understand why I need accept UserInformation as an argument.
Why is it better?
Do I need to create a new constructor for this or call a function with random parameters?